kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala

```diff
@@ -113,7 +132,29 @@ class FlinkProcessBuilder(
           flinkExtraJars += s"$hiveConfFile"
         }
 
-        val customFlinkConf = conf.getAllWithPrefix(FLINK_CONF_PREFIX, "")
+        val externalProxyUserConf: Map[String, String] = if (proxyUserEnable) {
+          // FLINK-31109: Flink only supports hadoop proxy user when delegation tokens fetch
```

Comment:
For tickets that applied to recent versions, let's leave the fixed version too. e.g.
```
// FLINK-31109 (1.17.0): ...
```